bin/tictactoe

```diff
@@ -1,3 +1,52 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+puts "Welcome to Tic Tac Toe!"
+puts "How many human players are there? (0, 1, 2): "
+number_of_players = gets.strip.to_i
+
+until number_of_players.between?(0,2)
+  puts "Please enter 0, 1, or 2:"
+  number_of_players = gets.strip.to_i
+end
+
+first_player = 0
+unless number_of_players == 0
+  puts "Which player should be X and go first?: "
+  first_player = gets.strip.to_i
+  until first_player.between?(1,2)
+    puts "Please enter 1 or 2: "
+    first_player = gets.strip.to_i
+  end
+end
+
+player1 = nil
+player2 = nil
+
+if number_of_players == 0
+  player1 = Computer.new("X")
+  player2 = Computer.new("O")
+elsif number_of_players == 1
+  if first_player == 1
+    player1 = Human.new("X")
+    player2 = Computer.new("O")
+  else
+    player1 = Computer.new("X")
+    player2 = Human.new("O")
+  end
+elsif number_of_players == 2
+  if first_player == 1
+    player1 = Human.new("X")
+    player2 = Human.new("O")
+  else
+    player1 = Human.new("O")
+    player2 = Human.new("X")
+  end
+end
+
+board = Board.new
+game = Game.new(player1, player2, board)
+
+game.play
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.